numpy/lib/histograms.py

```diff
@@ -318,9 +318,15 @@ def _get_bin_edges(a, bins, range, weights):
         raise ValueError('`bins` must be 1d, when an array')
 
     if n_equal_bins is not None:
+        # gh-10322 forces us to request this now to avoid inconsistency
+        bin_type = np.result_type(first_edge, last_edge, a)
+        if np.issubdtype(bin_type, np.integer):
+            bin_type = np.result_type(bin_type, float)
```

Comment:
What is the result type here with Python `float`? Might add a note.

Reply:
It seems to be `float` no matter what.

Reply:
So, we can just write `bin_type = float`? Seems reasonable. (Or `None`, since `linspace` will take care.)

Reply:
This seems more flexible in case we add an `int128` that only safely casts into a quad double or something in the distant future.

Reply:
I would just let `linspace` deal with that case, i.e., set `bin_type = None` here (or at `float32` to the one above). Obviously, not a big deal though.